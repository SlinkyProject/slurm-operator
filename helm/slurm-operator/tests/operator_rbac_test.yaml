# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
---
suite: test operator rbac
templates:
- operator/rbac.yaml
chart:
  version: 1.2.3
  appVersion: 1.2.3
release:
  name: test-release
  namespace: test-namespace

tests:
- it: manifest should match snapshot
  asserts:
  - matchSnapshot: {}

- it: should not render when disabled
  set:
    operator:
      enabled: false
  asserts:
  - hasDocuments:
      count: 0

- it: should not render when serviceAccount.create is false
  set:
    operator:
      serviceAccount:
        create: false
  asserts:
  - hasDocuments:
      count: 0
