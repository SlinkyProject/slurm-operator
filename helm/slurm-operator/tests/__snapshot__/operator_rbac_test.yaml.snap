manifest should match snapshot:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: slurm-operator
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: slurm-operator-1.2.3
      name: test-release-slurm-operator
      namespace: test-namespace
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: test-release-slurm-operator
      namespace: test-namespace
    rules:
      - apiGroups:
          - ""
        resources:
          - configmaps
          - events
          - persistentvolumeclaims
          - pods
          - secrets
          - services
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - nodes
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - pods/status
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - apps
        resources:
          - controllerrevisions
          - deployments
          - statefulsets
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - slinky.slurm.net
        resources:
          - accountings
          - controllers
          - loginsets
          - nodesets
          - restapis
          - tokens
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - slinky.slurm.net
        resources:
          - accountings/finalizers
          - controllers/finalizers
          - loginsets/finalizers
          - nodesets/finalizers
          - restapis/finalizers
          - tokens/finalizers
        verbs:
          - update
      - apiGroups:
          - slinky.slurm.net
        resources:
          - accountings/status
          - controllers/status
          - loginsets/status
          - nodesets/status
          - restapis/status
          - tokens/status
        verbs:
          - get
          - patch
          - update
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: slurm-operator
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: slurm-operator-1.2.3
      name: test-release-slurm-operator
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: test-release-slurm-operator
    subjects:
      - kind: ServiceAccount
        name: test-release-slurm-operator
        namespace: test-namespace
